community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/MergeIntoAcceptanceTest.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.neo4j.cypher.ExecutionEngineFunSuite
+
+class MergeIntoAcceptanceTest extends ExecutionEngineFunSuite{
+
+  test("ON CREATE with update one property") {
+    //given
+    createNode("A")
+    createNode("B")
+
+    //when
+    val update = execute("""MATCH (a {name:'A'}), (b {name:'B'})
+      |MERGE (a)-[r:TYPE]->(b) ON CREATE SET r.name = 'foo'""".stripMargin)
+
+    //then
+    update should use("Merge(Into)")
+    val res = execute("MATCH ()-[r:TYPE]->() RETURN extract(key IN keys(r)| key + '->' + r[key]) as keyValue")
```

Comment:
I want to make sure that I check all keys are added since some of the `SET` operations remove existing properties and some are additive. This was the best I I could think of, basically I want to do `toMap` on `r` and check all keys and values. I am open to suggestions how to do this better.